### PR TITLE
Creating jsonparser class

### PR DESCRIPTION
**Pull request by NotCampeur**
Resolve issues:
- #48 
- #30 
- #17
- #64

### CODE REVIEW

File: includes/spk_basic_functions.hpp [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	 */
+	std::vector<std::wstring> getStringSplit(std::wfstream& p_file, const std::wstring& p_delim, const int& p_expectedSize = -1);
+
+	/**
+	 * @brief Calculates the positive modulo of the given integer and divisor.
+	 * @param p_integer The integer value.
+	 * @param p_dividor The divisor.
+	 * @return The positive modulo result.
+	 */
+	int positive_modulo(int p_integer, int p_dividor);
```
Comment (EreliaCorp) [inline includes/spk_basic_functions.hpp:75]: Wrong nomination, my bad

File: src/application/modules/windows/spk_API_module.cpp
Comment (EreliaCorp) [inline src/application/modules/windows/spk_API_module.cpp:1]: Line 112 : we need to remove this

Comment (EreliaCorp) [reply src/application/modules/windows/spk_API_module.cpp:1]: Once remove, the include for iostream is not necessary

File: src/miscellaneous/JSON/spk_JSON_file.cpp
Comment (EreliaCorp) [inline src/miscellaneous/JSON/spk_JSON_file.cpp:1]: We should split this file into multiples, like spk_json_file_number / spk_json_file_string and spk_json_file, cause 550 lines is quite long !

File: src/miscellaneous/JSON/spk_JSON_file.cpp [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+#include <algorithm>
+#include <cwctype>
+
+#include "spk_basic_functions.hpp"
+
+namespace spk
+{
+	namespace JSON
+	{
+		static std::wstring _handleEscapeSequence(const std::wstring& p_fileContent)
```
Comment (EreliaCorp) [inline src/miscellaneous/JSON/spk_JSON_file.cpp:13]: no command for this one ? =p

File: src/miscellaneous/JSON/spk_JSON_object.cpp [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+					tmp.second->_content.index() == 2)
+					p_os << std::endl;
+
+				p_os << *(tmp.second);
+				if (&tmp != &(*map.rbegin()))
+					p_os << L',';
+				p_os << std::endl;
+			}
+			--_indent;
+			p_os << std::setw(_indent * _indentSize) << L"" << L"}";
```
Comment (EreliaCorp) [inline src/miscellaneous/JSON/spk_JSON_object.cpp:249]: Why the L"" ?

**Review by EreliaCorp — CHANGES_REQUESTED** [review]
Need to awnser question and perform a rebase

File: includes/data_structure/spk_data_buffer.hpp [hunk 4]
```diff
@@ -27,6 +27,28 @@ namespace spk
...
+    const DataBuffer& _deserializeString(StringType& str) const
+    {
+        str.resize(this->get<size_t>());
+        for (size_t i = 0; i < str.size(); i++)
+        {
+            *this >> str[i];
+        }
+        return *this;
+    }
+
```
Comment (EreliaCorp) [inline includes/data_structure/spk_data_buffer.hpp:51]: Not needed anymore

File: includes/miscellaneous/spk_JSON_file.hpp
Comment (EreliaCorp) [inline includes/miscellaneous/spk_JSON_file.hpp:1]: File to remove

File: src/application/modules/spk_widget_module.cpp [hunk 5]
```diff
@@ -27,6 +27,7 @@ namespace spk
 
 	void WidgetModule::render()
 	{
+		spk::cout << " ------ " << std::endl;
```
Comment (EreliaCorp) [inline src/application/modules/spk_widget_module.cpp:30]: To remove

**Review by EreliaCorp — CHANGES_REQUESTED** [review]
Apply comment before merge

File: src/miscellaneous/JSON/spk_JSON_object.cpp (hunk 3 above)
Comment (NotCampeur) [reply src/miscellaneous/JSON/spk_JSON_object.cpp:249]: The std::setw only apply to the next entry, the empty string is to apply the complete padding without an extra character

**Review by EreliaCorp — APPROVED** [review]

**Pull request merged**